scheduler/server/cluster_state.go

```diff
@@ -205,40 +205,42 @@ func (c *clusterState) taskCompleted(nodeId cluster.NodeId, flaky bool) {
 	c.numRunning--
 }
 
-func (c *clusterState) getNodeState(nodeId cluster.NodeId) (*nodeState, bool) {
+func (c *clusterState) getNodeState(nodeId cc.NodeId) (*nodeState, bool) {
 	ns, ok := c.nodes[nodeId]
 	return ns, ok
 }
 
 // update cluster state to reflect added and removed nodes
+// it will process at most DefaultClusterSize sets of updates with each call
 func (c *clusterState) updateCluster() {
 	defer c.stats.Latency(stats.SchedUpdateClusterLatency_ms).Time().Stop()
-	select {
-	case updates, ok := <-c.updateCh:
-		if !ok {
-			c.updateCh = nil
+	allUpdates := []cc.NodeUpdate{}
+LOOP:
+	for i := 0; i < common.DefaultClusterChanSize; i++ {
```

Comment:
The buffer sizes seem arbitrary - using the const here at 100, but internally we set a fetcher/cluster buffer of 1000 (and 100 for apiserver). Suggest using the const value in all of those places unless the other numbers were picked for a reason.

Reply:
- 100 here was just an arbitrary number
- 1000 for scheduler vs 100 for apiserver in internal scoot was based on worker updates being batched at 1000 workers.